src/engine/Service.h

```diff
@@ -60,6 +60,10 @@ class Service : public Operation {
           GetTsvFunction getTsvFunction = sendHttpOrHttpsRequest,
           std::shared_ptr<QueryExecutionTree> siblingTree = nullptr);
 
+  inline void setSiblingTree(std::shared_ptr<QueryExecutionTree> siblingTree) {
```

Comment:
The inline doesn't do anything here. 
And add a short docstring, the semantics of a sibling tree should be documented.